packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -69,22 +69,38 @@ export default createRule<[], MessageIds>({
     const services = getParserServices(context);
     const checker = services.program.getTypeChecker();
 
-    function isPromiseCatchAccess(node: TSESTree.Expression): boolean {
-      if (
-        !(
-          node.type === AST_NODE_TYPES.MemberExpression &&
-          isStaticMemberAccessOfValue(node, 'catch')
-        )
-      ) {
+    type ArgumentIndexToCheck = 0 | 1;
+    function getPromiseMethodInfo(node: TSESTree.Expression):
+      | {
+          argumentIndexToCheck: ArgumentIndexToCheck;
+          messageData: Record<'method' | 'append', string>;
+        }
+      | false {
+      if (node.type !== AST_NODE_TYPES.MemberExpression) {
+        return false;
+      }
+      const methodsToCheck = [
+        { method: 'catch', append: '' },
+        { method: 'then', append: ' rejection' },
+      ] as const;
+      const argumentIndexToCheck = methodsToCheck.findIndex(({ method }) =>
```

Comment:
I agree that the `-1` and the type assertion were a little bit clunky!
I preferred to have all of the options for `then` and `catch` consolidated rather than divided into an array and a `switch` like you had it. Based on this and a couple other of your comments (narrowing and return type), I felt that it might make most sense to inline the function. However, let me know if you don't like it and I'm happy to do something more like what you're recommending here.